processor/add_orchestrator_metadata/cluster_metadata_provider.go

```diff
@@ -15,34 +15,48 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package add_cluster_id
+package add_orchestrator_metadata
 
 import (
 	"context"
+	"github.com/elastic/elastic-agent-autodiscover/kubernetes/metadata"
+	agentconfig "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	k8s "k8s.io/client-go/kubernetes"
 )
 
 type ClusterHelper interface {
-	ClusterId() string
+	GetClusterMetadata() ClusterMetadata
 }
 
-type clusterHelper struct {
-	clusterId string
+type ClusterMetadataProvider struct {
+	metadata ClusterMetadata
+	logger   *logp.Logger
 }
 
-func newClusterHelper(client k8s.Interface) (ClusterHelper, error) {
+type ClusterMetadata struct {
+	clusterId   string
+	clusterName string
+}
+
+func newClusterMetadataProvider(client k8s.Interface, cfg *agentconfig.C, logger *logp.Logger) (ClusterHelper, error) {
 	clusterId, err := getClusterIdFromClient(client)
 	if err != nil {
 		return nil, err
 	}
-	return &clusterHelper{clusterId: clusterId}, nil
+
+	clusterIdentifier, err := metadata.GetKubernetesClusterIdentifier(cfg, client)
```

Comment:
Nice that you can re-use the baseline code that is used in Beats and Agent :). 
How to do you plan to support more providers? Will you push to the `metadata.GetKubernetesClusterIdentifier` directly?
Would that help if we isolate completely the `add_orchestrator_metadata` in a common library?

cc: @gizas @ruflin

Reply:
Hi @ChrsMark, thanks for commenting and bringing up some good points for this discussion.
>How do you plan to support more providers? Will you push to the metadata.GetKubernetesClusterIdentifier directly?

I haven't thought it through yet, but I am thinking maybe using an adapter pattern or something similar. That is, instead of calling the  `metadata.GetKubernetesClusterIdentifier` directly I will wrap this method with an interface and will call the required implementation based on auto-discover/configuration.
Another option is, as you suggested, to enhance the `metadata.GetKubernetesClusterIdentifier` function directly.

> Would that help if we isolate completely the add_orchestrator_metadata in a common library?

If you think that other beats will use it, I don't mind moving it to the beats repository or any other repository.

Reply:
Hey! I think the functionality of getting the k8s orchestrator metadata should live only in https://github.com/elastic/elastic-agent-autodiscover/tree/main/kubernetes (where `metadata.GetKubernetesClusterIdentifier` actually is). So I would lean towards implementing the support for various providers there. Also see https://github.com/elastic/beats/issues/33081 for the plans to add support for other providers within the Agent's library (cc: @gizas @tetianakravchenko )

In addition I believe the processor you are introducing here would make sense to live in a common place so as to be available to other beats too like Heartbeat etc. Imagine that I want to only do system level monitoring on a k8s cluster and for this I use Metricbeat and `system` module. If I want to add cluster metadata at the moment this is not possible. Your implementation here would solve that so actually it's a really cool feature for all of the Beats.

Reply:
@ChrsMark 

So just to make sure I understand, you suggesting adding this functionality to the `Beats` repository?

Reply:
Yeap, but I'm still not 100% sure if that would be the right way to go. My concern is about how this would work if someone enables the `add_kubernetes_metadata` processor at the same time. 
We would need to investigate the code organisation more to answer this.

So in this regard I would suggest to move forward this PR without taking into account these concerns and comment on https://github.com/elastic/beats/issues/33081 to raise an action item to revisit how we can organize the various implementations better.